src/http/request.cr

```diff
@@ -98,6 +98,19 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    host = @headers["Host"]?
+    return unless host
+    index = host.index(":")
+    index ? host[0...index] : host
```

Comment:
@jhass Unfortunately, `.` may also act as a delimiter in literal IPv6 addresses, even mixed. See the RFC referenced above.